code/game/bioscans.dm

```diff
@@ -126,13 +131,17 @@ GLOBAL_DATUM_INIT(bioscan_data, /datum/bioscan_data, new)
 /datum/bioscan_data/proc/qm_bioscan(variance = 2)
 	/// Adjust the randomness there so everyone gets the same thing
 	var/fake_marines_on_ship = max(0, marines_on_ship + rand(-variance, variance))
+	var/metalhive_hosts = "[fake_marines_on_ship ? "approximately [fake_marines_on_ship]":"no"]"
+	var/plural = "[!fake_marines_on_ship || fake_marines_on_ship > 1 ? "s":""]"
+	var/metalhive_location = "[fake_marines_on_ship&&marine_ship_location?", including one in [marine_ship_location],":""]"
+	var/planet_hosts = "[marines_on_planet ? "[marines_on_planet]" : "none"]"
+	var/planet_location = "[marines_on_planet && marine_planet_location ? ", including one in [marine_planet_location]" : ""]"
+
+	log_game("BIOSCAN: Queen Mother bioscan completed.")
+	log_game("BIOSCAN: To my children and their Queen. I sense [metalhive_hosts] host[plural] in the metal hive [metalhive_location] and [planet_hosts] scattered elsewhere[planet_location].")
+
 	/// Shout it at everyone
 	for(var/mob/current_mob as anything in GLOB.living_xeno_list)
 		current_mob << sound(get_sfx("queen"), wait = 0, volume = 50)
 		to_chat(current_mob, SPAN_XENOANNOUNCE("The Queen Mother reaches into your mind from worlds away."))
-		var/metalhive_hosts = "[fake_marines_on_ship ? "approximately [fake_marines_on_ship]":"no"]"
-		var/plural = "[!fake_marines_on_ship || fake_marines_on_ship > 1 ? "s":""]"
-		var/metalhive_location = "[fake_marines_on_ship&&marine_ship_location?", including one in [marine_ship_location],":""]"
-		var/planet_hosts = "[marines_on_planet ? "[marines_on_planet]" : "none"]"
-		var/planet_location = "[marines_on_planet && marine_planet_location ? ", including one in [marine_planet_location]" : ""]"
 		to_chat(current_mob, SPAN_XENOANNOUNCE("To my children and their Queen. I sense [metalhive_hosts] host[plural] in the metal hive [metalhive_location] and [planet_hosts] scattered elsewhere[planet_location]."))
```

Comment:
Both to_chat messages here should just be accessing a var so a new string doesn't have to be created for every mob in the loop. It would also allow you to re-use that string also for the logged message.

You did move some of the work out, but the span wrapping and combination are still being done per mob.